lib/urlwatch/reporters.py

```diff
@@ -625,3 +635,113 @@ def submit_to_slack(self, webhook_url, text):
+
+        if cfg['minimal']:
+            for job_state in self.report.get_filtered_job_states(self.job_states):
+                pretty_name = job_state.job.pretty_name()
+                location = job_state.job.get_location()
+                if pretty_name != location:
+                    location = '%s (%s)' % (pretty_name, location)
+                yield '* ' + ': '.join((job_state.verb.upper(), location))
+            return
+
+        summary = []
+        details = []
+        for job_state in self.report.get_filtered_job_states(self.job_states):
+            summary_part, details_part = self._format_output(job_state)
+            summary.extend(summary_part)
+            details.extend(details_part)
+
+        if summary:
+            yield from ('%d. %s' % (idx + 1, line) for idx, line in enumerate(summary))
+            yield ''
+
+        if show_details:
+            yield from details
+
+        if summary and show_footer:
+            yield from ('--- ',
+                        '%s %s, %s  ' % (urlwatch.pkgname, urlwatch.__version__, urlwatch.__copyright__),
+                        'Website: %s  ' % (urlwatch.__url__,),
+                        'watched %d URLs in %d seconds' % (len(self.job_states), self.duration.seconds))
+
+    def _format_content(self, job_state):
+        if job_state.verb == 'error':
+            return job_state.traceback.strip()
+
+        if job_state.verb == 'unchanged':
+            return job_state.old_data
+
+        if job_state.old_data in (None, job_state.new_data):
+            return None
+
+        return self.unified_diff(job_state)
+
+    def _format_output(self, job_state):
+        summary_part = []
+        details_part = []
+
+        pretty_name = job_state.job.pretty_name()
+        location = job_state.job.get_location()
+        if pretty_name != location:
+            location = '%s (%s)' % (pretty_name, location)
+
+        pretty_summary = ': '.join((job_state.verb.upper(), pretty_name))
+        summary = ': '.join((job_state.verb.upper(), location))
+        content = self._format_content(job_state)
+
+        summary_part.append(pretty_summary)
+
+        details_part.append('### ' + summary)
+        if content is not None:
+            details_part.extend(('', '```', content, '```', ''))
+        details_part.extend(('', ''))
+
+        return summary_part, details_part
+
+
+class MatrixReporter(MarkdownReporter):
+    """Custom Matrix reporter"""
+    MAX_LENGTH = 4096
+
+    __kind__ = 'matrix'
+
+    def submit(self):
+        homeserver_url = self.config['homeserver']
+        access_token = self.config['access_token']
+        room_id = self.config['room_id']
+
+        body_markdown = '\n'.join(super().submit())
+
+        if not body_markdown:
+            logger.debug('Not calling Matrix API (no changes)')
+            return
+
+        if len(body_markdown) > self.MAX_LENGTH:
+            body_markdown = body_markdown[:self.MAX_LENGTH]
```

Comment:
Maybe add an ellipsis or something (be sure to subtract the additional characters from `MAX_LENGTH` so that the resulting string is still below `MAX_LENGTH` characters) so that it's obvious that some content was cut off, otherwise if somebody is super unlucky it will be cut off and they won't know that it was cut off and miss a notification.

Reply:
I simply copied the code from other reporters. The fix to the problem you've described should probably be applied to all such occurences uniformly.